src/lib.rs

```diff
@@ -552,10 +627,79 @@ impl Debug for Ext4 {
 #[cfg(test)]
 mod tests {
     use super::*;
+    use test_util::load_test_disk1;
+
+    fn block_read_error(
+        block_index: u64,
+        offset_within_block: u32,
+        read_len: usize,
+    ) -> Corrupt {
+        Corrupt::BlockRead {
+            block_index,
+            offset_within_block,
+            read_len,
+        }
+    }
+
+    /// Test that reading from the first 1024 bytes of the file fails.
+    #[test]
+    fn test_read_from_block_first_1024() {
+        let fs = load_test_disk1();
+        let mut dst = vec![0; 1024];
+        assert_eq!(
+            fs.read_from_block(0, 1023, &mut dst)
+                .unwrap_err()
+                .as_corrupt()
+                .unwrap(),
+            &block_read_error(0, 1023, 1024),
+        );
+    }
```

Comment:
If I comment out the corresponding check in the function, this test still passes (it emits an error when we try to read past the end of the block). Perhaps:

        let mut dst = vec![0; 256];
        assert_eq!(
            fs.read_from_block(0, 512, &mut dst)
                .unwrap_err()
                .as_corrupt()
                .unwrap(),
            &block_read_error(0, 512, 256),
        );

Reply:
Ah, but better to test the boundaries, so:

    let mut dst = vec![0; 1];
    assert_eq!(
        fs.read_from_block(0, 1023, &mut dst)
            .unwrap_err()
            .as_corrupt()
            .unwrap(),
        &block_read_error(0, 1023, 1),
    );

?

Reply:
done